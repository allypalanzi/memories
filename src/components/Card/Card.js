import React from 'react'
import PropTypes from 'prop-types';

import styles from './Card.scss'

const Card = (props) => {
  const { value, onClick, flipped, matched } = props;
  const classList = [styles.card];
  if (flipped || matched) {classList.push(styles.flipped)};
  if (matched) { classList.push(styles.matched)};

  return (
    <button disabled={matched} onClick={onClick} className={classList.join(' ')}>
    <div className={styles.icon_container}>
      <div className={flipped || matched ? `${styles.icon} ${styles.visible}` : styles.icon }>
        {value}
        </div>
      <div className={flipped || matched ? `${styles.icon}` : `${styles.icon} ${styles.visible}` }>
        âœ¨
      </div>
    </div>
    </button>
  );
}

Card.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func,
  flipped: PropTypes.bool,
  matched: PropTypes.bool
}

export default Card;
