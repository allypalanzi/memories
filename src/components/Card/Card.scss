.card {
  background: #b6f8ff;
  padding: 50px 0;
  flex-basis: 20%;
  margin: 2%;
  color: #3b3b3b;
  font-size: 30px;
  border-radius: 2%;
  border: 1px solid darken(#b6f8ff, 10%);
  line-height: 1;
  transition: all 100ms ease-in-out;
  cursor: pointer;
  &:hover:not(:disabled),
  &:focus:not(:disabled) {
    transform: scale(1.1) rotate(-2deg);
    transition: all 100ms ease-in-out;
    box-shadow: 2px 2px 19px 2px #dedddd;
  }
}

.matched {
  background-color: #f3f3f3;
  border: 1px solid darken(#f3f3f3, 10%);
  color: lighten(#3b3b3b, 50%);
}
