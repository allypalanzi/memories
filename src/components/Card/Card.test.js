import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Card from './Card'

describe('<Card />', () => {
  const onClick = jest.fn()
  const props = {
    onClick,
    flipped: false,
    matched: false,
    value: 'ðŸ‘‹',
  }
  const renderedComponent = shallow(<Card {...props} />)

  it('renders correctly', () => {
    expect(toJson(renderedComponent)).toMatchSnapshot()
  })

  it('calls onClick when card is clicked', () => {
    renderedComponent.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })

  // TODO: tests for styles based on props
})
