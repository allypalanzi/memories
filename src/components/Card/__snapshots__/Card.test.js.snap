// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<button
  className="card"
  onClick={[Function]}
>
  <div
    className="icon_container"
  >
    <div
      className="icon"
    >
      ðŸ‘‹
    </div>
    <div
      className="icon visible"
    >
      <span
        aria-label="sparkle emoji"
        role="img"
      >
        âœ¨
      </span>
    </div>
  </div>
</button>
`;
